docs/notes/2.27.x.md

```diff
@@ -38,6 +38,7 @@ In [the `[ruff]` subsystem](https://www.pantsbuild.org/2.27/reference/subsystems
 
 ### Plugin API changes
 
+* Process execution can now specify a minimum amount of cpu cores required for a process to run via the `concurrency` field. `concurrency_available` is deprecated in favor of `concurrency`
```

Comment:
Let me know what tweaks you want to the release notes. I could copy what I have in my pr description or feel free to suggest edits directly